repo2docker/app.py

```diff
@@ -453,24 +453,39 @@ def push_image(self):
         client = docker.APIClient(version="auto", **kwargs_from_env())
         # Build a progress setup for each layer, and only emit per-layer
         # info every 1.5s
+        progress_layers = {}
         layers = {}
         last_emit_time = time.time()
-        for line in client.push(self.output_image_spec, stream=True):
-            progress = json.loads(line.decode("utf-8"))
-            if "error" in progress:
-                self.log.error(progress["error"], extra=dict(phase="failed"))
-                raise docker.errors.ImageLoadError(progress["error"])
-            if "id" not in progress:
-                continue
-            if "progressDetail" in progress and progress["progressDetail"]:
-                layers[progress["id"]] = progress["progressDetail"]
-            else:
-                layers[progress["id"]] = progress["status"]
-            if time.time() - last_emit_time > 1.5:
-                self.log.info(
-                    "Pushing image\n", extra=dict(progress=layers, phase="pushing")
-                )
-                last_emit_time = time.time()
+        for chunk in client.push(self.output_image_spec, stream=True):
+            for line in chunk.splitlines():
```

Comment:
You might be wondering why this extra iteration was added. When I tested this locally, each chunk could be *multiple* JSON lines, split by `\r\n`. So I added splitting lines to the iteration. I'm not sure why this hasn't come up on mybinder.org. I'm using `docker 3.7.1`

Reply:
Should we add this comment as a comment to the code? I'd vote yes